cloud/ec2_fleet.go

```diff
@@ -171,17 +171,24 @@ func (m *ec2FleetManager) GetInstanceStatus(ctx context.Context, h *host.Host) (
 	}
 	defer m.client.Close()
 
-	instance, err := m.getInstance(ctx, h)
+	instance, err := m.client.GetInstanceInfo(ctx, h.Id)
 	if err != nil {
-		return status, errors.Wrapf(err, "can't get instance status")
+		grip.Error(message.WrapError(err, message.Fields{
+			"message":       "error getting instance info",
+			"host":          h.Id,
+			"host_provider": h.Distro.Provider,
+			"distro":        h.Distro.Id,
+		}))
+		return status, errors.Wrap(err, "error getting instance info")
 	}
 
+	if instance.State == nil || instance.State.Name == nil || *instance.State.Name == "" {
+		return status, errors.New("state name is missing")
+	}
 	status = ec2StatusToEvergreenStatus(*instance.State.Name)
 	if status == StatusRunning {
 		// cache instance information so we can make fewer calls to AWS's API
-		if err = cacheHostData(ctx, h, instance, m.client); err != nil {
-			return status, errors.Wrapf(err, "can't update host '%s'", h.Id)
-		}
+		grip.Warning(errors.Wrapf(cacheHostData(ctx, h, instance, m.client), "can't update host '%s'", h.Id))
```

Comment:
I think it's worth logging at an error level. Also use message.fields so that it's easier to search

Reply:
Done.